app-drawer/app-drawer.html

```diff
@@ -42,75 +44,69 @@
 
   <style>
 
-    :host {
-      position: absolute;
+    #drawerContent {
+      position: fixed;
       top: 0;
-      right: auto;
       bottom: 0;
       left: 0;
       width: 256px;
-      overflow: hidden;
-      background-color: var(--app-drawer-background-color, white);
-      visibility: hidden;
+      background-color: #FFF;
 
-      -webkit-transform: translate3d(-100%, 0, 0);
-      transform: translate3d(-100%, 0, 0);
+      -webkit-transform: translate(-100%, 0);
+      transform: translate(-100%, 0);
 
-      -webkit-transition-property: -webkit-transform, visibility;
-      transition-property: transform, visibility;
+      -webkit-transition-property: -webkit-transform;
+      transition-property: transform;
 
       -webkit-transition-duration: var(--app-drawer-duration, 0.2s);
       transition-duration: var(--app-drawer-duration, 0.2s);
 
-      -webkit-transition-timing-function: var(--app-drawer-timing-function, linear);
-      transition-timing-function: var(--app-drawer-timing-function, linear);
-    }
+      -webkit-transition-timing-function: var(--app-drawer-timing-function, ease);
+      transition-timing-function: var(--app-drawer-timing-function, ease);
 
-    :host([position=right]) {
-      right: 0;
-      left: auto;
-      -webkit-transform: translate3d(100%, 0, 0);
-      transform: translate3d(100%, 0, 0);
+      @apply(--app-drawer-styles);
     }
 
-    :host([position=top]) {
-      top: 0;
+    :host[position=right] #drawerContent {
+      left: auto;
       right: 0;
-      bottom: auto;
-      left: 0;
-      width: auto;
-      height: 256px;
-      -webkit-transform: translate3d(0, -100%, 0);
-      transform: translate3d(0, -100%, 0);
+
+      -webkit-transform: translate(100%, 0);
+      transform: translate(100%, 0);
```

Comment:
`translate3d` for own painting layer (not `will-change` for browser compatibility), here and everywhere.